app/domain/authentication/authn_oidc/authenticate_id_token/authenticate.rb

```diff
@@ -73,6 +73,7 @@ def conjur_username
 
           conjur_username = id_token_attributes[id_token_username_field]
           raise IdTokenFieldNotFoundOrEmpty, id_token_username_field unless conjur_username.present?
+          raise BlacklistedOidcUser, conjur_username if blacklisted_oidc_user?(conjur_username)
```

Comment:
Are there plans to blacklist other users? Otherwise, since we're only blacklisting the admin user, I would suggest being more specific here and name the error something like `AdminAuthenticationAttempt`. We could also rename the method being called `is_admin?`.

Reply:
will do. i had this more general for future use like we have in authn-ldap.

```
      # admin should only be able to login through plain Conjur authn
      def blacklisted_ldap_user?(login)
        login == 'admin'
      end
```

Reply:
I would say make it non-generic for now. Seems like we may never need to blacklist other users as admin is a very special case.